MediaAdminBundle/Form/Type/OrchestraSiteForFolderChoiceType.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace OpenOrchestra\MediaAdminBundle\Form\Type;
+
+use Symfony\Component\Form\FormBuilderInterface;
+use OpenOrchestra\ModelInterface\Model\SiteInterface;
+use OpenOrchestra\ModelInterface\Repository\SiteRepositoryInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use OpenOrchestra\MediaAdminBundle\Form\DataTransformer\EmbedSiteToSiteIdTransformer;
+
+/**
+ * Class OrchestraSiteForFolderChoiceType
+ */
+class OrchestraSiteForFolderChoiceType extends AbstractType
+{
+    protected $siteRepository;
+    protected $tokenStorage;
+    protected $embedSiteToSiteTransformer;
+
+    /**
+     * @param SiteRepositoryInterface $siteRepository
+     */
+    public function __construct(SiteRepositoryInterface $siteRepository, TokenStorageInterface $tokenStorage, EmbedSiteToSiteIdTransformer $embedSiteToSiteIdTransformer)
+    {
+        $this->siteRepository = $siteRepository;
+        $this->tokenStorage = $tokenStorage;
+        $this->embedSiteToSiteIdTransformer = $embedSiteToSiteIdTransformer;
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array                $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['embed']) {
+            $builder->addModelTransformer($this->embedSiteToSiteIdTransformer);
+        }
+    }
+
+    /**
+     * @param OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults(
+            array(
+                'embed' => false,
+                'choices' => $this->getChoices()
+            )
+        );
+    }
+
+    /**
+     * @return array
+     */
+    public function getChoices()
```

Comment:
I agree with you, it was protected until I made unit tests. I need to test it as it contains the specific logic of the formtype, but being protected it can't :/
Do you have an alternative ?